libsolidity/interface/CompilerStack.cpp

```diff
@@ -403,16 +402,12 @@ bool CompilerStack::parse()
 		}
 	}
 
-	if (m_stopAfter <= Parsed)
-		m_stackState = Parsed;
-	else
-		m_stackState = ParsedAndImported;
 	if (Error::containsErrors(m_errorReporter.errors()))
-		m_hasError = true;
+		return false;
 
+	m_stackState = (m_stopAfter <= Parsed ? Parsed : ParsedAndImported);
 	storeContractDefinitions();
-
-	return !m_hasError;
+	return true;
 }
```

Comment:
This change and the one in `analyze()` change behavior of `CompilerStack` because `Parsed` or `AnalysisSuccessful` will never be reached in presence of errors like they used to. I don't think there are any user-visible consequences of this other than the behavior of `--error-recovery` (which is experimental) and #14328, which already has a changelog entry. Therefore I'm not adding a separate changelog entry for this change.